examples/cache.rb

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require "bundler/setup"
+require "zk_recipes"
+require "zk"
+require "zk-server"
+
+begin
+  ZK_PORT = (27183..28_000).detect { |port| !system("nc -z localhost #{port}") }
+
+  ZK::Server.run do |c|
+    c.client_port = ZK_PORT
+    c.force_sync = false
+    c.snap_count = 1_000_000
+    c.max_session_timeout = 5_000 # ms
+  end
+
+  zk = ZK.new("localhost:#{ZK_PORT}")
+  zk.rm_rf("/test")
+  zk.mkdir_p("/test")
+
+  # A cache that creates its own ZK client:
+
+  cache = ZkRecipes::Cache.new(host: "localhost:#{ZK_PORT}", timeout: 10, zk_opts: { timeout: 5 }) do |z|
+    z.register("/test/boom", "goat")
+    z.register("/test/foo", 1) { |raw_value| raw_value.to_i * 2 }
+    z.register_children("/test/groupA", ["/test/foo"]) do |children|
+      children.map do |child|
+        path = "/test/#{child}"
+        unless cache.registered_runtime_path?(path) || cache.registered_path?(path)
+          cache.register_runtime(path, "")
+        end
+        path
+      end
+    end
```

Comment:
Thoughts on this as an alt?:

```ruby
children
  .map { |child| "/test/#{child}" }
  .select { |path| cache.registered_runtime_path?(path) || cache.registered_path?(path) }
  .each { |path| cache.register_runtime(path, "") }
```

Reply:
I can't quite do that, need the map to return the whole list, but I can do mostly that (I think):

```ruby
      children
        .map { |child| "/test/#{child}" }
        .each { |path| cache.register_runtime(path, "") if cache.registered_runtime_path?(path) || cache.registered_path?(path) }
```